cpp/lib/ClientConfigParser.cpp

```diff
@@ -0,0 +1,315 @@
+  std::string derivedConfigPath = "";
+  if (!in_configFilePath.empty())
+  {
+    // 1. Try config file if it was passed in
+    derivedConfigPath = in_configFilePath;
+    CXX_LOG_INFO("sf", "ClientConfigParser", "loadClientConfig",
+      "Using client configuration path from a connection string: %s", in_configFilePath.c_str());
+  }
+  else if (const char* clientConfigEnv = sf_getenv_s(SF_CLIENT_CONFIG_ENV_NAME.c_str(), envbuf, sizeof(envbuf)))
+  {
+    // 2. Try environment variable SF_CLIENT_CONFIG_ENV_NAME
+    derivedConfigPath = clientConfigEnv;
+    CXX_LOG_INFO("sf", "ClientConfigParser", "loadClientConfig",
+      "Using client configuration path from an environment variable: %s", clientConfigEnv);
+  }
+  else
+  {
+    // 3. Try DLL binary dir
+    std::string binaryDir = getBinaryPath();
+    std::string binaryDirFilePath = binaryDir + SF_CLIENT_CONFIG_FILE_NAME;
+    if (checkFileExists(binaryDirFilePath))
+    {
+      derivedConfigPath = binaryDirFilePath;
+      CXX_LOG_INFO("sf", "ClientConfigParser", "loadClientConfig",
+        "Using client configuration path from binary directory: %s", binaryDirFilePath.c_str());
+    }
+    else
+    {
+#if defined(WIN32) || defined(_WIN64)
+      // 4. Try user home dir
+      std::string homeDir = sf_getenv_s("USERPROFILE", envbuf, sizeof(envbuf));
+      if (homeDir.empty())
+      {
+        // USERPROFILE is empty, try HOMEDRIVE and HOMEPATH
+        std::string homeDriveEnv = sf_getenv_s("HOMEDRIVE", envbuf, sizeof(envbuf));
+        std::string homePathEnv = sf_getenv_s("HOMEPATH", envbuf, sizeof(envbuf));
+        if (!homeDriveEnv.empty() && !homePathEnv.empty())
+        {
+          homeDir = homeDriveEnv + homePathEnv;
+        }
+      }
+      std::string homeDirFilePath = homeDir + PATH_SEP + SF_CLIENT_CONFIG_FILE_NAME;
+      if (checkFileExists(homeDirFilePath))
+      {
+        derivedConfigPath = homeDirFilePath;
+        CXX_LOG_INFO("sf", "ClientConfigParser", "loadClientConfig",
+          "Using client configuration path from home directory: %s", homeDirFilePath.c_str());
+      }
+#else
+      // 4. Try user home dir
+      if (const char* homeDir = sf_getenv_s("HOME", envbuf, sizeof(envbuf)))
+      {
+        std::string homeDirFilePath = std::string(homeDir) + PATH_SEP + SF_CLIENT_CONFIG_FILE_NAME;
+        if (checkFileExists(homeDirFilePath))
+        {
+          derivedConfigPath = homeDirFilePath;
+          CXX_LOG_INFO("sf", "ClientConfigParser", "loadClientConfig",
+            "Using client configuration path from home directory: %s", homeDirFilePath.c_str());
+        }
+      }
+#endif
+    }
+  }
+  if (!derivedConfigPath.empty())
+  {
+    parseConfigFile(derivedConfigPath, out_clientConfig);
+  }
+}
+
+// Private =====================================================================
+////////////////////////////////////////////////////////////////////////////////
+bool ClientConfigParser::checkFileExists(const std::string& in_filePath)
```

Comment:
We have boost in libsfclient I think better to use boost::filesystem::exists

Reply:
std::filesystem isn't available on osx10.14 and boost has a debug assertion on 32-bit windows with MT build

Reply:
I think using boost filesystem was able to fix the issue